python/drned_xmnr/op/base_op.py

```diff
@@ -184,24 +193,26 @@ def run_with_trans(self, callback, write=False, db=_ncs.RUNNING):
             mp = maapi.Maapi()
             return callback(mp.attach(self.uinfo.actx_thandle))
 
-    def extend_timeout(self, timeout_extension):
-        dp.action_set_timeout(self.uinfo, timeout_extension)
+    def extend_timeout(self):
+        extension = self.device_timeout + 2*TIMEOUT_MARGIN
```

Comment:
Totally agree, it's far from obvious; I didn't find a good place where to explain this so let's do it here:

There are several layers, each having it's own timeout. It is desirable and less disruptive that if the lower layer is able to somehow abort whatever is going on because it takes too long, the higher level waits for it to do so, so the higher layer's timeout should be at least somewhat bigger than that of the lower layer, to avoid nasty race conditions. At the same time, it is confusing to have two or three different timeout parameters that all need to be configured with meaningful values.

So this is that: there is only one parameter, `/devices/device/read-timeout`, used by all three layers, with increasing margin:

1. the lowest layer, DrNED, needs to use the timeout value to abort waiting for ncs_cli response if that does not come
2. XMNR uses `timeout+margin` to wait for DrNED's output; this might not be needed, if we could be sure that DrNED *always* aborts if it takes too long, but we cannot, or at least I am not sure we can rely on DrNED in this respect
3. finally, NCS is repeatedly told to wait another `timeout+2*margin` for the action to complete; we need to do that, since otherwise NCS might simply abort the action.